fsspec/caching.py

```diff
@@ -510,33 +576,37 @@ def _fetch(self, start, stop):
         return out + super()._fetch(start, stop)
 
 
-class UpdatableLRU:
+class UpdatableLRU(Generic[P, T]):
     """
     Custom implementation of LRU cache that allows updating keys
 
     Used by BackgroudBlockCache
     """
 
-    CacheInfo = collections.namedtuple(
-        "CacheInfo", ["hits", "misses", "maxsize", "currsize"]
-    )
+    class CacheInfo(NamedTuple):
+        hits: int
+        misses: int
+        maxsize: int
+        currsize: int
 
-    def __init__(self, func, max_size=128):
-        self._cache = collections.OrderedDict()
+    def __init__(self, func: Callable[P, T], max_size: int = 128) -> None:
+        self._cache: OrderedDict[Any, T] = collections.OrderedDict()
         self._func = func
         self._max_size = max_size
         self._hits = 0
         self._misses = 0
         self._lock = threading.Lock()
 
-    def __call__(self, *args):
+    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> T:
+        if kwargs:
```

Comment:
Why reject kwargs if actually we pass them on to _func, below?

Reply:
This is just for backwards compatibility with the previous code (which doesn't take `kwargs`). Happy to also allow `kwargs` to be passed if you'd prefer!

Unfortunately, `ParamSpec` is quite limited and there's no way to say `args-only` for ParamSpec AFAICT -- we could also juts do `*args: Any` without type-checking the # of kind of args.